### ISSUE TITLE
Improve prediction algorithm

### ISSUE THREAD

**Issue by levinsamuel** [levinsamuel/boisbot#4]
This is the guide I am following for now: https://machinelearningmastery.com/text-generation-lstm-recurrent-neural-networks-python-keras/

The prediction is not that great. See what I can do to make it better.

**Comment by levinsamuel** [comment]
With the latest model, here is the seed and prediction:

SEED:
> DEBUG:predict:Pattern chosen is: ke, as in, nit-picking. it doesnt really make sense, because how are they nit-picking? not sure. any ways

PREDICTION:
> suarsed corkd hase a serer of like a bunch of copald team iards__tekee hersets tie sooe and they can ae a pittoe a bunch of people wery

**Comment by levinsamuel** [comment]
Here's his suggestions copied from the article:

> 10 Extension Ideas to Improve the Model
Below are 10 ideas that may further improve the model that you could experiment with are:
> 
> - Predict fewer than 1,000 characters as output for a given seed.
> - Remove all punctuation from the source text, and therefore from the models’ vocabulary.
> - Try a one hot encoded for the input sequences.
> - Train the model on padded sentences rather than random sequences of characters.
> - Increase the number of training epochs to 100 or many hundreds.
> - Add dropout to the visible input layer and consider tuning the dropout percentage.
> - Tune the batch size, try a batch size of 1 as a (very slow) baseline and larger sizes from there.
> - Add more memory units to the layers and/or more layers.
> - Experiment with scale factors (temperature) when interpreting the prediction probabilities.
> - Change the LSTM layers to be “stateful” to maintain state across batches.
> - Did you try any of these extensions? Share your results in the comments.

**Comment by levinsamuel** [comment]
I wonder if it's possible to do word-based predictions rather than character-based. Seems like that might require a massive amount of data though.
